cmd/goal/application.go

```diff
@@ -282,13 +327,44 @@ func parseAppArg(arg appCallArg) (rawValue []byte, parseErr error) {
 	return
 }
 
-func parseAppInputs(inputs appCallInputs) (args [][]byte, accounts []string, foreignApps []uint64, foreignAssets []uint64) {
+func translateBoxRefs(input []boxRef, foreignApps []uint64) []transactions.BoxRef {
+	output := make([]transactions.BoxRef, len(input))
+	for i, tbr := range input {
+		rawName, err := tbr.name.raw()
+		if err != nil {
+			reportErrorf("Could not decode box name %s: %v", tbr.name, err)
+		}
+
+		index := uint64(0)
+		if tbr.appId != 0 {
```

Comment:
What happens if the `appId` equals its own explicit, i.e. not `0`, app ID? I'm guessing it should be translated to 0, but I'm not sure if this happens in this loop.

Reply:
Yes, that's a good idea.  Otherwise the caller would have to supply its own app-id in the foreign-app array. Thanks.